### PR TITLE
docs: more updates to Banner accessibility docs

### PR DESCRIPTION
**Pull request by khiga8**
Following up on https://github.com/primer/design/pull/724, I made additional updates to improve readability by updating the heading structure and rephrasing things!

[Preview doc 🔍 ](https://primer-0fe4ab137c-26441320.drafts.github.io/components/banner)

### CODE REVIEW

File: content/components/banner.mdx [hunk 1]
```diff
@@ -10,34 +10,44 @@ import ComponentLayout from '~/src/layouts/component-layout'
...
-  * For communicating non-critical information (such as a success), a live region announcement is sufficient in **most** scenarios. Keep in mind that a live region announcement can sometimes fail to fire and can easily be missed, so it should be reserved for use on non-critical information.
-* Does the Banner contain an action?
-  * If the Banner contains an action, placing focus on the Banner is preferable so that a screen reader user can immediately interact with the action without having to manually locate it.
-* Is there already a live region on the page that can be utilized?
-  * [Dynamic live regions can have some challenges](https://github.com/github/accessibility/blob/main/docs/coaching-recommendations/toast-flash-banner/accessible-banner-prototype.md#challenges-with-dynamically-inserted-live-region), so this method requires ample testing.
+[Dynamic live regions can have some challenges](https://github.com/github/accessibility/blob/main/docs/coaching-recommendations/toast-flash-banner/accessible-banner-prototype.md#challenges-with-dynamically-inserted-live-region), so this method requires ample testing.
```
Comment (khiga8) [inline content/components/banner.mdx:50]: [Dynamic live regions can have some challenges (GitHub Staff only)](https://github.com/github/accessibility/blob/main/docs/coaching-recommendations/toast-flash-banner/accessible-banner-prototype.md#challenges-with-dynamically-inserted-live-region)

Comment (khiga8) [reply content/components/banner.mdx:50]:
```suggestion
[Dynamic live regions can have some challenges (GitHub Staff Only)](https://github.com/github/accessibility/blob/main/docs/coaching-recommendations/toast-flash-banner/accessible-banner-prototype.md#challenges-with-dynamically-inserted-live-region), so this method requires ample testing.
```

File: content/components/banner.mdx [hunk 2]
```diff
@@ -10,34 +10,44 @@ import ComponentLayout from '~/src/layouts/component-layout'
...
 
-### Focus management vs. Live region announcement
+### Feedback Banners
 
-Banner can be used in one of two ways:
+Banners that are used to communicate feedback require extra accessibility considerations to ensure they are surfaced immediately to all users including those using assistive technologies. This relates to [WCAG 4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html).
+
+Example scenarios that require extra considerations include:
+
+* When a user submits a form, an error banner is shown with a list of form errors that needs to be corrected.
```
Comment (lindseywild) [inline content/components/banner.mdx:26]:
```suggestion
* When a user submits a form, an error banner is shown with a list of form errors that need to be corrected.
```

**Review by lindseywild — APPROVED** [review]

File: content/components/banner.mdx [hunk 3]
```diff
@@ -10,34 +10,52 @@ import ComponentLayout from '~/src/layouts/component-layout'
...
+
+Here are some questions to consider when deciding which method to use:
+
+#### How critical is the information?
+
+Critical information (such as an error) should be made as discoverable as possible. Though moving focus to the Banner can be considered more disruptive than a live region announcement, it's more important that a user is able to discover the Banner so they can take an appropriate action. In most critical scenarios, using focus management is the best approach.
+
+On the other hand, for communicating newly shown non-critical information (such as a success), a live region announcement is sufficient in **most** scenarios. A live region announcement can sometimes fail to fire and can easily be missed, so it should be reserved for use on non-critical information where there are no consequences if a user misses it.
+
+#### Where is a user's focus when the banner appears?
```
Comment (khiga8) [inline content/components/banner.mdx:44]: I realized I accidentally deleted this question in my previous PR, so I added it back.

**Pull request merged**